### ISSUE TITLE
[MAYA-128598] Fatal error if delete USD node with USD layer Editor opened

### ISSUE THREAD

**Issue by kimuss** [Autodesk/maya-usd#2985]
**Describe the bug**
When I try to delete a USD stage node from outliner with USD Layer Editor opened, Maya crash.

**Steps to reproduce the behavior:**

1. Go to Create
2. Click on Universal Scene Description--> Stage From File
3. Open USD Layer Editor panel
4. Delete Stage node on Outliner
5. Fatal error

**Specs:**
 - OS & version Windows 11 22H2
 - Maya version Maya 2023.3
 - Maya USD Plugin 0.22

**Additional context**

Same fatal error on Maya 2022.4
If I try to delete the Stage node with USD Layer Editor panel closed, everything works fine

**Comment by Lucas3Dspain** [comment]
I have seen this happening since 2022.4 too, It doesn't happen every for me.

Same in Maya 2024 (mayaUSD 0.22)

**Comment by santosd** [comment]
@kimuss and @Lucas3Dspain Maya 2024 was just released today along with Maya USD 0.23.0. I tested with those versions and I am not seeing the issue. Could you test that and see if you are still seeing this?

**Comment by Lucas3Dspain** [comment]
It still happens, I tried to narrow down when because it doesn't happen all the time(at least for me, I think it has to do with the sublayers)

Looks like when I have multiple sublayers and several of them have unsaved edits, it crashes when deleting the proxyShape

1. open a stage that has sublayers. 
2. Add some edits to the root layer
3. Add some edits to any sublayer
4. optional, mute other sublayer
5. Open the UsdLayerEditor, 
6. delete USD proxy shape from the outliner

if you skip step 5 or close the window, it doesn't crash.

or simply, in a new scene open the shot.usd as a proxy shape, do step 5 and 6 and you will get a  crash
[mayaUSD_layerEditorCrash.zip](https://github.com/Autodesk/maya-usd/files/11106895/mayaUSD_layerEditorCrash.zip)

If you open layout.usd instead of  shot.usd, the issue doesn't happen, because this layer doesn't have sublayers I think.

**Comment by santosg87** [comment]
@santosd I think we found a reproducible case here - slightly different workflow, as you need 1 more step, but pretty similar workflow, and a consistent crash:

1. Go to Create
2. Click on Universal Scene Description--> Stage From File
3. Open USD Layer Editor panel
4. Add a sublayer under your anonymous layer
5. Delete Stage node on Outliner
6. Fatal error


we have an issue logged internally as MAYA-128598

**Comment by santosg87** [comment]
@Lucas3Dspain hi! :D it seems we were typing together!

**Comment by kimuss** [comment]
you are right, it happen with anonymous layer and for me also when delete stage node on Outliner from network file.usd stage (layer editor always opened).
Anyway on Maya 2024 I can not create a sublayer on sessionLayer anymore,  like i could that in 2023:

![sessionlayer](https://user-images.githubusercontent.com/54888290/228775747-167b79d7-13f0-433c-afa9-3ac05e7fb5b2.jpg)

**Comment by santosd** [comment]
Excellent! Thank you all. We will take a look at it and hopefully get a fix in soon.
